dev-docs/dependency-upgrades.adoc

```diff
@@ -0,0 +1,64 @@
+= Dependency upgrades
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+Solr has lots of 3rd party dependencies, defined mainly in `versions.props`.
+Keeping them up-to-date is crucial for a number of reasons:
+
+* minimizing the risk of critical CVE vulnerabilities by staying on a recent and supported version
+* avoiding "dependency hell", that can arise from falling too far behind
+
+Read the `help/dependencies.txt` file for an in-depth explanation of how gradle is deployed in Solr, using
+https://github.com/palantir/gradle-consistent-versions[Gradle consistent-versions] plugin.
+
+== Manual dependency upgrades
+In order to upgrade a dependency, you need to run through a number of steps:
+
+1. Identify the available versions from e.g. https://search.maven.org[Maven Central]
+2. Update the version in `versions.props` file
+3. Run `./gradlew --write-locks` to re-generate `versions.lock`. Note that this may cause a cascading effect where
+   the locked version of other dependencies also change.
+4. Run `./gradlew updateLicenses` to re-generate SHA1 checksums of the new jar files.
+5. Once in a while, a new version of a dependency will transitively bring in brand-new dependencies.
+   You'll need to decide whether to keep or exclude them. See `help/dependencies.txt` for details.
+
+== Renovate bot Pull Requests
+A member of the Solr community operates a Github bot running https://github.com/renovatebot/renovate[Renovate], which
```

Comment:
Instead of all this text on renovatebot in our docs, we should put them all in the external repository and maybe add a one liner here referring to it.

Reply:
At the risk of being flamed, I don't know why you are so against the word "renovatebot" in our repo?   If I see PR's showing up in Solr, I would love to see some documentation about it...   I think more documentation is a good thing.

Reply:
I didn't say I'm against the "word renovatebot" in our repo. If you read carefully, I said that so much verbose explanation can reside in the external project and a concise one liner describing renovatebot with a link to the external project should be better.

Reply:
>  If I see PR's showing up in Solr, I would love to see some documentation about it

If the bot can add a link to the documentation page (in its own repo) along with each PR, then this concern can be addressed.

Reply:
Let's gain some experience and then revisit docs.

Reply:
@chatman I removed the "Maintaining the bot" paragraph from the dev-doc. Agree it is enough to mention it over at the bot README.